x/harvest/keeper/liquidation.go

```diff
@@ -0,0 +1,307 @@
+		return sdkerrors.Wrapf(types.ErrBorrowNotLiquidatable, "borrowed %s <= borrowable %s", totalBorrowedUSDAmount, totalBorrowableUSDAmount)
+	}
+
+	// Sending coins to auction module with keeper address getting % of the profits
+	borrow, _ := k.GetBorrow(ctx, borrower)
+	err := k.SeizeDeposits(ctx, keeper, liqMap, deposits, borrowBalances, depositDenoms, borrowDenoms)
+	if err != nil {
+		return err
+	}
+
+	k.DeleteBorrow(ctx, borrow)
+
+	// TODO: these 'oldDeposits' can still have funds in them. Where should we send the extra balance?
+	for _, oldDeposit := range deposits {
+		k.DeleteDeposit(ctx, oldDeposit)
+	}
+
+	return nil
+}
+
+// SeizeDeposits seizes a list of deposits and sends them to auction
+func (k Keeper) SeizeDeposits(ctx sdk.Context, keeper sdk.AccAddress, liqMap map[string]LiqData,
+	deposits []types.Deposit, borrowBalances sdk.Coins, dDenoms, bDenoms []string) error {
+
+	// Seize % of every deposit and send to the keeper
+	aucDeposits := sdk.Coins{}
+	for _, deposit := range deposits {
+		mm, _ := k.GetMoneyMarket(ctx, deposit.Amount.Denom)
+		keeperReward := mm.KeeperRewardPercentage.MulInt(deposit.Amount.Amount).TruncateInt()
+		keeperCoin := sdk.NewCoin(deposit.Amount.Denom, keeperReward)
+
+		// Send keeper their reward
+		err := k.supplyKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, keeper, sdk.NewCoins(keeperCoin))
+		if err != nil {
+			return err
+		}
+
+		// Add remaining deposit coin to aucDeposits
+		aucDeposits = aucDeposits.Add(sdk.NewCoin(deposit.Amount.Denom, deposit.Amount.Amount.Sub(keeperReward)))
+	}
+
+	// Build map to hold deposit coin USD valuations
+	totalRemainingDepositedUSDAmount := sdk.ZeroDec()
+	dUsdMap := make(map[string]sdk.Dec)
+	for _, deposit := range aucDeposits {
+		dData := liqMap[deposit.Denom]
+		dCoinUsdValue := sdk.NewDecFromInt(deposit.Amount).Quo(sdk.NewDecFromInt(dData.conversionFactor)).Mul(dData.price)
+		totalRemainingDepositedUSDAmount = totalRemainingDepositedUSDAmount.Add(dCoinUsdValue)
+		dUsdMap[deposit.Denom] = dCoinUsdValue
+	}
+
+	// Build map to hold borrow coin USD valuations
+	totalBorrowedUSDAmount := sdk.ZeroDec()
+	bUsdMap := make(map[string]sdk.Dec)
+	for _, bCoin := range borrowBalances {
+		bData := liqMap[bCoin.Denom]
+		bCoinUsdValue := sdk.NewDecFromInt(bCoin.Amount).Quo(sdk.NewDecFromInt(bData.conversionFactor)).Mul(bData.price)
+		totalBorrowedUSDAmount = totalBorrowedUSDAmount.Add(bCoinUsdValue)
+		bUsdMap[bCoin.Denom] = bCoinUsdValue
+	}
+
+	// The % by which the lot must be larger than the borrow
+	ltv := totalBorrowedUSDAmount.Quo(totalRemainingDepositedUSDAmount)
+
+	err := k.StartAuctions(ctx, deposits[0].Depositor, bDenoms, dDenoms, borrowBalances, aucDeposits, ltv, liqMap, bUsdMap, dUsdMap)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// StartAuctions attempts to start auctions for seized assets
+func (k Keeper) StartAuctions(ctx sdk.Context, borrower sdk.AccAddress, borrowDenoms, depositDenoms []string,
```

Comment:
I think we can get this down to one double loop that will create the minimum possible number of auctions with an optimization.

Optimization: Make the denom lists sorted by USD value for borrows and deposits. That way, we will always start an auction for a full borrow amount if possible. 

Pseudocode below

```
for bDenom in borrowDenoms:
  maxLotSize := bUsdMap[bDenom].Quo(ltv) // note we will be updating this value as we loop over depositDenoms

  for dDenom in depositDenoms: 
    if maxLotSize.IsZero():
      break // exit out of the loop if we have cleared the full amount
    if dUsdMap[dDenom].GTE(maxLotSize): // we can start an auction for the whole barrow amount
      // start an auction with the whole borrow amount as bid and maxLotSize as lot
      // ...

      maxLotSize = sdk.ZeroDec()
    else: // we can only start an auction for the partial barrow amount
      maxBid = dUsdMap[dDenom].Mul(ltv)
      // start an auction with maxBid as bid and the whole deposit amount as lot
      // ..

      // update variables to account for partial auction
      bUsdMap[bDenom] = bUsdMap[bDenom].Sub(maxBid)
      maxLotSize = bUsdMap[bDenom].Quo(ltv) 
```

Reply:
Implemented an initial version of this in https://github.com/Kava-Labs/kava/pull/731/commits/a952fbcef836ea0407e644e6dae78444f1f76284